lib/units/unit.dart

```diff
@@ -222,49 +228,35 @@ abstract class Unit extends SpriteAnimationGroupComponent<UnitAnimationState>
     attacking = enemy;
   }
 
-  bool moveToBlock(final Block block) {
+  void moveToBlock(final Block block) {
     attacking = null; // stop attacking
     final occupiedBlocks = map.occupiedBlocks;
-    var width = 1;
-    var height = 1;
-    final x = block.x;
-    final y = block.y;
-    Block? nextBlock = block;
-
-    for (;;) {
-      if (map.validBlock(nextBlock)) {
-        target = nextBlock;
-        occupiedBlocks.remove(reservedBlock);
-        occupiedBlocks.add(nextBlock!);
-        reservedBlock = nextBlock;
-        return true;
-      }
-      final topLeft = Block(x - width, y - height);
-      final bottomRight = Block(x + width, y + height);
-      final topRight = Block(bottomRight.x, topLeft.y);
-      final bottomLeft = Block(topLeft.x, bottomRight.y);
-      final blocks = _blocksBetween(topLeft, topRight)
-        ..addAll(_blocksBetween(topRight, bottomRight))
-        ..addAll(_blocksBetween(bottomRight, bottomLeft))
-        ..addAll(_blocksBetween(bottomLeft, topLeft));
-      nextBlock = blocks.firstWhereOrNull(map.validBlock);
-      width++;
-      height++;
-    }
-  }
-
-  Set<Block> _blocksBetween(Block a, Block b) {
-    final blocks = <Block>{};
-    final minX = min(a.x, b.x);
-    final maxX = max(a.x, b.x);
-    final minY = min(a.y, b.y);
-    final maxY = max(a.y, b.y);
-    for (var x = minX; x <= maxX; x++) {
-      for (var y = minY; y <= maxY; y++) {
-        blocks.add(Block(x, y));
-      }
-    }
-    return blocks;
+    final currentBlock = this.block;
+    final startingOffset = Offset(
+      currentBlock.x.toDouble(),
+      currentBlock.y.toDouble(),
```

Comment:
block.toVector2().toOffset()

Reply:
This method is already too expensive, we could add an extension to block that gives an offset.

Reply:
Right this is not in the map, so it should be fine, but doesn't really add readability to first convert it to a vector.